core/src/vaults.ts

```diff
@@ -381,6 +391,51 @@ const getLatestVault = async (network: string, walletAddress: string) => {
     return new BigNumber(lastHex._hex).toNumber();
 };
 
+export const openVaultWithProxiedContractAndDrawDebt = async (
+    network: string,
+    proxyAddress: string,
+    collateralType: CollateralType,
+    collateralAmount: BigNumber,
+    debtAmountDai: BigNumber,
+    proxyType = 'CROPPER'
+) => {
+    const config = getCollateralConfigByType(collateralType);
+    giveAllowanceToAddress(network, config.symbol, proxyAddress, collateralAmount);
+
+    const signer = await getSigner(network);
+    const proxyContract = new ethers.Contract(proxyAddress, DS_PROXY, signer);
+    const method = getMethodSignature('openLockGemAndDraw(address,address,address,bytes32,uint256,uint256)');
+    const jugContractAddress = await getContractAddressByName(network, 'MCD_JUG');
+    const joinContractCollateralAddress = await getContractAddressByName(
+        network,
+        getJoinNameByCollateralType(collateralType)
+    );
+    const joinContractDaiAddress = await getContractAddressByName(network, 'MCD_JOIN_DAI');
+    const args = [
+        jugContractAddress,
+        joinContractCollateralAddress,
+        joinContractDaiAddress,
+        ethers.utils.formatBytes32String(collateralType),
+        collateralAmount.shiftedBy(WAD_NUMBER_OF_DIGITS).toFixed(),
+        debtAmountDai.shiftedBy(DAI_NUMBER_OF_DIGITS).toFixed(),
+    ];
+    const typesArray = ['address', 'address', 'address', 'bytes32', 'uint256', 'uint256'];
+    const encodedArgs = ethers.utils.defaultAbiCoder.encode(typesArray, args);
+    const transactionData = method + encodedArgs.substring(2);
+    const target = (await getContract(network, `PROXY_ACTIONS_${proxyType}`)).address;
+    const transaction = await proxyContract['execute(address,bytes)'](target, transactionData);
+    const events = await extractEventFromTransaction(
+        transaction,
+        'NewCdpRegistered(address,address,uint256)',
+        new ethers.utils.Interface(CDP_REGISTRY)
+    );
```

Comment:
You basically provide ABI 3 times: 
1. It comes with `transaction`
2. The `NewCdpRegistered(address,address,uint256)` is the ABI of the event you need to get
3. The interface itself `new ethers.utils.Interface(CDP_REGISTRY)`

Initially I suggested to use ABI that is provided with transaction, but it turns out it's not feasible due to the key event being emitted by the other contract. So instead you can use `'NewCdpRegistered(address,address,uint256)'` to create interface and pass it to decode specific event that you need.

You might even replace second parameter with `NewCdpRegistered(address sender,address owner,uint256 cdp)` to retrieve values under the named key `args.cdp`. In other words:

```ts
const events = await extractEventFromTransaction(
    transaction,
    'NewCdpRegistered(address sender, address owner, uint256 cdp)'
);
```
then, inside `extractEventFromTransaction`
```
const contractInterface = new ethers.utils.Interface('NewCdpRegistered(address sender, address owner, uint256 cdp)')
```
While keeping the rest of the code the same


Nit: the function should be called `extractEventsFromTransaction` as it returns an array and might make `eventABI` parameter optional

Reply:
This suggestion gets the same problem as https://github.com/ethers-io/ethers.js/issues/1233 where incomplete interface specification results in problems with parsing the transaction logs. Hence i resided with providing signature and complete abi separately. Do you have the better suggestion than `try-catching`?

Reply:
> receives "what to look for" and "how to parse"

No, you provide "what to look for + types" and "the whole interface of the contract", while `what to look for + types` in theory is enough to parse what you need. I don't see a problem with using try/catch to get only desired events.

Anyway, I think you can keep it as is if you want to add the interface of the whole contract when you need just a single event, but by suggestion is to simplify it. And it's a nit.

Reply:
1. i've tried to implement the try-catch as in the linked issue. It won't "just work" - the interface's `parseLog` and `decodeEventLog` do not result in any return (assuming that i did not mess up something obvious & primitive)
2. given that the parsing by the incomplete interface results in the errors instead of array of logs - i also get the impression that the intention there is to use the parsing logic with the complete abi.

Due to the 2 above points + the intent of resulting code to be "simple" - i would side with the current solution still.